packages/core/src/extensions/extension-loader/file-system-provisioner-store/file-system-provisioner-store.ts

```diff
@@ -4,36 +4,49 @@
  */
 
 import { SHA256 } from "crypto-js";
-import { action, makeObservable, observable } from "mobx";
-import type { BaseStoreDependencies } from "../../../common/base-store/base-store";
-import { BaseStore } from "../../../common/base-store/base-store";
+import { action, observable } from "mobx";
+import type { BaseStore } from "../../../common/base-store/base-store";
 import type { LensExtensionId } from "../../lens-extension";
-import { getOrInsertWithAsync, toJS } from "../../../common/utils";
+import { getOrInsertWithAsync, object } from "../../../common/utils";
 import type { EnsureDirectory } from "../../../common/fs/ensure-dir.injectable";
 import type { JoinPaths } from "../../../common/path/join-paths.injectable";
 import type { RandomBytes } from "../../../common/utils/random-bytes.injectable";
+import type { CreateBaseStore } from "../../../common/base-store/create-base-store.injectable";
 
-interface FSProvisionModel {
-  extensions: Record<string, string>; // extension names to paths
+interface FileSystemProvisionerStoreModel {
+  extensions: Partial<Record<string, string>>; // extension names to paths
 }
 
-interface Dependencies extends BaseStoreDependencies {
+interface Dependencies {
```

Comment:
Why not `extends BaseStoreDependencies` anymore?

Reply:
Actually we specifically don't pass the whole of `this.dependencies` when we call `this.dependencies.createBaseStore()`.

The reason for this is the point of this PR. It is to make `BaseStore` compostable instead of inheritable. That way if `BaseStore` needs to add or remove a dependency we don't have to add a `di.inject()` call in all the places where subclasses are instantiated but instead only in the `createBaseStoreInejctable`.